### PR TITLE
changes from array to List

### PR DESCRIPTION
**Pull request by Dheerajr444**
### Description
[Describe what this change achieves]

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [x] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/cluster/metadata/Metadata.java [hunk 1]
```diff
@@ -291,12 +291,12 @@ static Custom fromXContent(XContentParser parser, String name) throws IOExceptio
         final Map<String, IndexMetadata> indices,
         final Map<String, IndexTemplateMetadata> templates,
         final Map<String, Custom> customs,
-        String[] allIndices,
-        String[] visibleIndices,
-        String[] allOpenIndices,
-        String[] visibleOpenIndices,
-        String[] allClosedIndices,
-        String[] visibleClosedIndices,
+        List<String> allIndices,
```
Comment (vikasvb90) [inline server/src/main/java/org/opensearch/cluster/metadata/Metadata.java:294]: What is this change intended for? Can you update the PR description and may be raise a logical cut of the change in the PR if it is a part of a feature?

**Review by vikasvb90 — CHANGES_REQUESTED** [review]

**Comment by sandeshkr419** [comment]
Hi @Dheerajr444 

Are you trying to touch the same code piece and same logical changes as done in https://github.com/opensearch-project/OpenSearch/pull/14723 & discussed in https://github.com/opensearch-project/OpenSearch/issues/8647

Please do comment on issues to let others know that you are working on it so that redundant work can be avoided.
